lib/Remotery.h

```diff
@@ -661,6 +691,23 @@ RMT_API void _rmt_BeginMetalSample(rmtPStr name, rmtU32* hash_cache);
 RMT_API void _rmt_EndMetalSample(void);
 #endif
 
+// Iterator
```

Comment:
Does this API compile and link when `RMT_ENABLED=0`? Prefix-underscores shouldn't be used here.

As an example:

```cpp
RMT_API void _rmt_LogText(rmtPStr text);
```

This is the function signature that is part of the Private interface and this is the macro which gets compiled-out when `RMT_ENABLED=0`:

```cpp
#define rmt_LogText(text)                                                           \
    RMT_OPTIONAL(RMT_ENABLED, _rmt_LogText(text))
```

Reply:
I have now added this, but I'm wondering if the use case is a bit different.
The client code will fail to compile on code like this if we simply omit the info:

```
error: expected expression
    const char* name = rmt_SampleGetName(ctx->rmt, sample);
```

I see two options:
1. Each such interface needs to produce "valid" values, which is counter intuitive when using the `RMT_OPTIONAL + RMT_ENABLED`
2. Letting the client control their code using `#ifdef RMT_ENABLED` explicitly around this particular piece of code.

Reply:
Yes, there's going to be a lot of errors if you implement some of them in terms of `RMT_OPTIONAL` as they need to return a value even when `RMT_ENABLED=0`.

One option is to return nothing and specify the output as an address parameter to the function call but that would then prevent you from initialising at the declaration point in C when multiple calls are involved.

Have you looked at using `RMT_OPTIONAL_RET`?

Reply:
I updated the api now to use the RMT_OPTIONAL+RMT_OPTIONAL_RET and the dump.c example builds and runs in `-DRMT_ENABLED=0` as well.